pkg/features/features.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package features
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+)
+
+var (
+	mu sync.Mutex
```

Comment:
I am not sure I like a global mutex... I understand it's used for  synchronizing calls to the functions below but if those functions need synchronization because they work on the same state/data I would prefer to encapsulate everything in a struct, mutex and all... even if we end up with the hated singleton :sweat:

Reply:
++ agreed, the mutex should be part of the struct.  I thought I fixed this already but I think it was the same situation in a different repo's PR or something.  Will fix here, thanks!

Reply:
Fixed in 312f63f50.  Let me know what you think.